web/api/wrapper.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+// WrapperCommand represents components necessary for OAuth2l request
+type WrapperCommand struct {
+	RequestType string
+	Args
+	Credential
+}
+
+// Args type used for unmarshalled JSON
+type Args map[string]interface{}
+
+// Credential type used for storing JSON-formatted credentials
+type Credential map[string]interface{}
+
+// Execute will capture output of OAuth2l CLI using command args
+func (wc WrapperCommand) Execute() (output string, err error) {
+	// combinedArgs used to represent command arguments in flattened array
+	args, err := combinedArgs(wc)
+
+	if err != nil {
+		return "", err
+	}
+
+	// Check if credential is provided to include in oauth2l command
+	if wc.Credential != nil {
+		// Symlink path to memory file
+		path, err := getCredentialPath(wc.Credential)
+
+		if err != nil {
+			return "", err
+		}
+
+		args = append(args, "--credentials", path)
+	}
+
+	// Execute command and capture output
+	command := exec.Command("oauth2l", args...)
+	byteBuffer, err := command.Output()
+
+	// Convert byteBuffer to string and remove newline character
+	output = strings.TrimSuffix(string(byteBuffer), "\n")
+	
+	return output, err
+}
+
+func combinedArgs(wc WrapperCommand) (combinedArgs []string, err error) {
+	combinedArgs = append(combinedArgs, wc.RequestType)
+
+	for flag, value := range wc.Args {
+		combinedArgs = append(combinedArgs, flag)
+		
+		// Assert args are of accepted types
+		switch value := value.(type) {
+		case []string:
+			combinedArgs = append(combinedArgs, value...)
```

Comment:
What does this type correspond to? It may be good to add a function comment regarding the accepted types and provide an example of each type.